tests/portForwardingServer.spec.ts

```diff
@@ -15,111 +15,134 @@
+        const wsEndpoint = await server.listen(0);
+        const service = await PlaywrightClient.connect({
+          wsEndpoint,
+          forwardPorts,
+        });
+        const playwright = service.playwright();
+        const browser = await playwright[browserName].launch(browserOptions);
+        return await browser.newPage();
+      }
+      const browserServer = await browserType.launchServer({
+        ...browserOptions,
+        _acceptForwardedPorts: acceptForwardedPorts
+      } as LaunchOptions);
+      browserServers.push(browserServer);
+      const browser = await browserType.connect({
+        wsEndpoint: browserServer.wsEndpoint(),
+        _forwardPorts: forwardPorts
+      } as ConnectOptions);
+      return await browser.newPage();
+    });
+    for (const browserServer of browserServers)
+      await browserServer.close();
+    for (const playwrightServer of playwrightServers)
+      await playwrightServer.close();
+  },
+});
 
+it.fixme(({ platform, browserName }) => platform === 'darwin' && browserName === 'webkit');
 it.skip(({ mode }) => mode !== 'default');
-it.fixme(({platform, browserName}) => platform === 'darwin' && browserName === 'webkit');
-
-let targetTestServer: http.Server;
-let port!: number;
-it.beforeAll(async ({}, test) => {
-  port = 30_000 + test.workerIndex * 4;
-  targetTestServer = http.createServer((req: http.IncomingMessage, res: http.ServerResponse) => {
-    res.end('<html><body>from-retargeted-server</body></html>');
-  }).listen(port);
-});
 
 it.beforeEach(() => {
   delete process.env.PW_TEST_PROXY_TARGET;
 });
 
-it.afterAll(() => {
-  targetTestServer.close();
-});
-
-it('should forward non-forwarded requests', async ({ browserType, browserOptions, server }, workerInfo) => {
-  process.env.PW_TEST_PROXY_TARGET = port.toString();
-  let reachedOriginalTarget = false;
-  server.setRoute('/foo.html', async (req, res) => {
-    reachedOriginalTarget = true;
-    res.end('<html><body>original-target</body></html>');
-  });
-  const browserServer = await browserType.launchServer({
-    ...browserOptions,
-    _acceptForwardedPorts: true
-  } as LaunchOptions);
-  const browser = await browserType.connect({
-    wsEndpoint: browserServer.wsEndpoint(),
-    _forwardPorts: []
-  } as ConnectOptions);
-  const page = await browser.newPage();
-  await page.goto(server.PREFIX + '/foo.html');
-  expect(await page.content()).toContain('original-target');
-  expect(reachedOriginalTarget).toBe(true);
-  await browserServer.close();
-});
-
-it('should proxy local requests', async ({ browserType, browserOptions, server }, workerInfo) => {
-  process.env.PW_TEST_PROXY_TARGET = port.toString();
-  let reachedOriginalTarget = false;
-  server.setRoute('/foo.html', async (req, res) => {
-    reachedOriginalTarget = true;
-    res.end('<html><body></body></html>');
+function startTestServer() {
+  const server = http.createServer((req: http.IncomingMessage, res: http.ServerResponse) => {
```

Comment:
Don't you have to await for `listening` callback?

Reply:
> This function is asynchronous
from their documentation. So better to do so.